plugins/inputs/statsd/running_stats.go

```diff
@@ -69,12 +82,35 @@ func (rs *RunningStats) AddValue(v float64) {
 		// Reached limit, choose random index to overwrite in the percentile array
 		rs.perc[rand.Intn(len(rs.perc))] = v
 	}
+
+	if len(rs.med) < rs.MedLimit {
+		rs.med = append(rs.med, v)
+	} else {
+		// Reached limit, start over
+		rs.med[rs.MedInsertIndex] = v
+		rs.MedInsertIndex = (rs.MedInsertIndex + 1) % rs.MedLimit
+	}
```

Comment:
I think my suggestion is a bit fragile... We probably should always update the index... What do you think?
```suggestion
	if len(rs.med) < rs.MedLimit {
		rs.med = append(rs.med, v)
	} else {
		// Reached limit, start over
		rs.med[rs.MedInsertIndex] = v
	}
	rs.MedInsertIndex = (rs.MedInsertIndex + 1) % rs.MedLimit
```

Reply:
Yeah good point! Changed and I even added a test to make sure MedInsertIndex and MedLimit are properly working.